core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -167,5 +169,18 @@ private static String pathContext(String tableLocation) {
 
       return resolvedContext;
     }
+
+    private static String computeHash(String fileName) {
+      Preconditions.checkState(fileName != null, "fileName cannot be null");
+      byte[] messageDigest =
+          HASH_FUNC.hashBytes(fileName.getBytes(StandardCharsets.UTF_8)).asBytes();
+
+      StringBuilder hash = new StringBuilder();
+      for (int i = 0; i < 8; ++i) {
```

Comment:
can we make 8 a static variable?

Reply:
is there any benefit in making it a variable?

Reply:
I guess from what Daniel replied below the answer is no and 8 is more than enough.

Reply:
+1, not required at the moment, though delta has this property called `randomPrefixLength` defaulting to `2` [code pointer](https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/DeltaConfig.scala#L436-L445) .if required, can take this as a follow-up pr.